tools/importer-rest-api-specs/components/testing/complete.go

```diff
@@ -49,9 +49,11 @@ func (tb TestBuilder) generateCompleteTest(dependencies *testDependencies) (*str
 func hasOptionalField(input map[string]resourcemanager.TerraformSchemaFieldDefinition, schemaModels map[string]resourcemanager.TerraformSchemaModelDefinition) bool {
 	for _, details := range input {
 		if details.ObjectDefinition.NestedObject != nil {
-			nestedModel := schemaModels[*details.ObjectDefinition.NestedObject.ReferenceName]
-			if hasOptionalField(nestedModel.Fields, schemaModels) {
-				return true
+			if details.ObjectDefinition.NestedObject.ReferenceName != nil {
```

Comment:
_minor_ but the rest of the codebase checks based on the Type of the (Nested) Object Definition (and if Type is a Reference then we require a ReferenceName) - so it'd be good to match that behaviour:

```suggestion
			if details.ObjectDefinition.NestedObject.Type == resourcemanager.ReferenceApiObjectDefinitionType {
```